jax/lax/lax.py

```diff
@@ -1115,11 +1115,14 @@ def _reduce_and(operand: Array, axes: Sequence[int]) -> Array:
 
 def reduce_window(operand: Array, init_value: Array, computation: Callable,
                   window_dimensions: Shape, window_strides: Sequence[int],
-                  padding: str) -> Array:
+                  padding: Union[str, Sequence[Tuple[int, int]]]) -> Array:
   """Wraps XLA's `ReduceWindow
   <https://www.tensorflow.org/xla/operation_semantics#reducewindow>`_
```

Comment:
This docstring is not quite accurate anymore, it seems. XLA reducewindow is documented to take padding as a string, without the option to pass (lo, hi) pairs.

Reply:
Is it actually important to expose the (lo, hi) API? In this case, `reduce_window` is more like the XLA `ReduceWindowWithGeneralPadding`.

Reply:
Yes, in general we always want to expose the full generality of XLA.

We already started using this functionality as a simplification in this PR in the implementation of the cumulative reductions on TPU.

There doesn't appear to be any documentation on ReduceWindowWithGeneralPadding in XLA; that seems to be a flaw in the XLA documentation.

In general I wouldn't call this operation `ReduceWindowWithGeneralPadding`. It's just `ReduceWindow` with fewer special cases and an API that more closely matches convolution. I don't want to mirror something that simply hasn't been cleaned up in XLA's API.

Reply:
Changed the link text to say `ReduceWindowWithGeneralPadding` even though the link still points to the only documentation there is.